### ISSUE TITLE
gatsby-plugin-offline

### ISSUE THREAD

**Issue by deadcoder0904** [gatsbyjs/gatsby#6830]
## Description

The error Netlify shows is as follows & the build fails after that -

The plugin "gatsby-plugin-offline@2.0.0-beta.5" is exporting a variable named "onServiceWorkerInstalled" which isn't an API. Perhaps you meant to export "registerServiceWorker"?

### Steps to reproduce

Just install `gatsby-plugin-manifest@next` & `gatsby-plugin-offline@next` & add the following config

```js

// gatsby-config.js

module.exports = {
    {
      resolve: `gatsby-plugin-manifest`,
      options: {
        name: `GatsbyJS`,
        short_name: `GatsbyJS`,
        start_url: "/",
        background_color: "#f7f0eb",
        theme_color: "#a2466c",
        display: "minimal-ui",
        icon: "./src/favicon.png" // This path is relative to the root of the site.
      }
    },
    `gatsby-plugin-offline`
};
```

### Environment

```
  System:
    OS: macOS High Sierra 10.13.6
    CPU: x64 Intel(R) Core(TM) i5-6360U CPU @ 2.00GHz
    Shell: 5.3 - /bin/zsh
  Binaries:
    Node: 10.6.0 - /usr/local/bin/node
    Yarn: 1.7.0 - /usr/local/bin/yarn
    npm: 6.1.0 - /usr/local/bin/npm
  Browsers:
    Chrome: 68.0.3440.75
    Safari: 11.1.2
  npmPackages:
    gatsby: 2.0.0-beta.55 => 2.0.0-beta.55 
    gatsby-image: ^2.0.0-beta.7 => 2.0.0-beta.7 
    gatsby-plugin-emotion: ^2.0.0-beta.3 => 2.0.0-beta.3 
    gatsby-plugin-favicon: ^3.1.2 => 3.1.2 
    gatsby-plugin-feed: ^2.0.0-beta.4 => 2.0.0-beta.4 
    gatsby-plugin-google-analytics: ^2.0.0-beta.3 => 2.0.0-beta.3 
    gatsby-plugin-manifest: ^2.0.2-beta.3 => 2.0.2-beta.3 
    gatsby-plugin-netlify: ^2.0.0-beta.4 => 2.0.0-beta.4 
    gatsby-plugin-netlify-cache: ^0.1.0 => 0.1.0 
    gatsby-plugin-nprogress: ^2.0.0-beta.4 => 2.0.0-beta.4 
    gatsby-plugin-offline: ^2.0.0-beta.5 => 2.0.0-beta.5 
    gatsby-plugin-react-helmet: 3.0.0-beta.4 => 3.0.0-beta.4 
    gatsby-plugin-sharp: ^2.0.0-beta.7 => 2.0.0-beta.7 
    gatsby-plugin-sitemap: ^2.0.0-beta.3 => 2.0.0-beta.3 
    gatsby-plugin-twitter: ^2.0.0-beta.3 => 2.0.0-beta.3 
    gatsby-remark-images: ^2.0.1-beta.9 => 2.0.1-beta.9 
    gatsby-source-filesystem: ^2.0.1-beta.9 => 2.0.1-beta.9 
    gatsby-transformer-remark: ^2.1.1-beta.4 => 2.1.1-beta.4 
    gatsby-transformer-sharp: ^2.1.1-beta.6 => 2.1.1-beta.6 
  npmGlobalPackages:
    gatsby-cli: 1.1.58
```
### File contents (if changed)

`gatsby-config.js`: N/A 
`package.json`: N/A 
`gatsby-node.js`: N/A 
`gatsby-browser.js`: N/A 
`gatsby-ssr.js`: N/A

**Comment by deadcoder0904** [comment]
@KyleAMathews I changed all my versions in `package.json` to use `@next` so it deployed but Service Worker (Manifest Plugin) & Offline Plugin is not working. 

Can you check the config above? Is there something I am doing wrong?

**Comment by deadcoder0904** [comment]
I'm also using `gatsby-plugin-favicon` so does it conflict with Manifest or Offline❓

**Comment by deadcoder0904** [comment]
I got it working by commenting `gatsby-plugin-favicon` & clearing build cache in Netlify since I'm using `gatsby-plugin-netlify-cache`

**Comment by deadcoder0904** [comment]
Hey @julien1619 Idk what's the use of `gatsby-plugin-favicon` if `gatsby-plugin-manifest` & `gatsby-plugin-offline` generates Favicons, installs Service Workers & is ready for offline support which in turn makes it a PWA?

I know `gatsby-plugin-favicon` generates variety of favicons so why not include it in `gatsby-plugin-manifest` itself so it generates all variety of favicons for all devices❓I think that's what it does best so integrating it IMO is the best thing for everyone I think. What do you say @KyleAMathews❓

**Comment by julien1619** [comment]
Actually, `gatsby-plugin-manifest` wasn't generating favicon last year, that's why I created this plugin. In April I've had some feature request to add manifest configuration to my plugin to reach `gatsby-plugin-manifest` features. I didn't have time to do that until this week, and now it's generating a fully configurable WebApp manifest. In the meantime it seems that they decided to add some favicon generation to `gatsby-plugin-manifest` and I wasn't aware of that until now.

The way I see it, we should do a comparison between the two plugins, they seem to do the same thing, and they seem to reach more or less the same feature when used with `gatsby-plugin-offline`, and see what we should do : merge it and deprecate one, keep separated with different goals, or .. ?

**Comment by deadcoder0904** [comment]
Yeah that's cool. I'll let you 2 talk now. I understood the differences now that they are both same 😂

**Comment by LeKoArts** [comment]
> and see what we should do : merge it and deprecate one, keep separated with different goals, or .. 

In my opinion one should be deprecated (probably `plugin-favicon` as manifest is more popular, already in the repo, included in many tutorials & starters) or you remove the manifest part and only do the favicon part (and place a note for usage of both plugins)

**Comment by julien1619** [comment]
I compared both and the current state is the following:
- Generated favicons: `gatsby-plugin-favicon` generates all possible favicons with some customizations available depending of the the type (Legacy ico, Standards, Apple Startup, Android, WebApp icons, ...), `gatsby-plugin-manifest` only generates some standards ones and the WebApp icons.
- Manifest customization: both offer more or less the same level of customization.
- Caching: `gatsby-plugin-favicon` has caching and does not generate icons again and has a cache bursting feature by appending a hash to the favicons folder (deactivable). This is not the case for `gatsby-plugin-manifest`. Actually, the lack of this feature allows the `gatsby-plugin-offline` to work smoothly with `gatsby-plugin-manifest` because some files are always at the same place (the manifest mainly).
- Popularity: Based on npm usage, 4515 downloads last week for `gatsby-plugin-favicon` and 7566 for `gatsby-plugin-manifest`.
- Library: `gatsby-plugin-manifest` is completely written from scratch (using sharp to resize icons), `gatsby-plugin-favicon` uses the underlying library `favicons` which is also used by a lot of standard webpack sites (35000+ downloads last week) that's why the number of supported favicons and manifests is larger in `gatsby-plugin-favicon`.

If I deprecate mine, I think we are losing some features, unless we backport it to `gatsby-plugin-manifest` with some changes needed on `gatsby-plugin-offline` if we keep cache-bursting too.

If I keep mine, I could add a parameter to disable manifest handling, but icons for the manifest should still be generated by `gatsby-plugin-manifest` because of the cache bursting that will prevent it from finding icons generated by `gatsby-plugin-favicon`, unless the user disables it. If someone wants to use the manifest from `gatsby-plugin-favicon` and remove `gatsby-plugin-manifest`, he could too.

What do you think?
